### PR TITLE
Feature 14 : Migration Versioning and CRON Job

### PR DESCRIPTION
**Pull request by Abhishek-Jain-1925**
Feature 7 : Migration Versioning and CRON Job

Workdone are as follows - 
1. Added Migration Versioning
2. Added CRON Jobs for Every Day Data Backup.
3. Added Column Profile-> Josh Joining Date.

### CODE REVIEW

File: internal/repository/profile_repository.go [hunk 1]
```diff
@@ -334,3 +335,70 @@ func (profileStore *ProfileStore) HandleTransaction(ctx context.Context, tx pgx.
...
+	fileName := fmt.Sprintf("%s%s-profile_builder.sql", backupDir, time.Now().Format("20060102"))
+
+	file, err := os.Create(fileName)
+	if err != nil {
+		zap.S().Errorw("Failed to create backup file", "error", err)
+		return
+	}
+	defer file.Close()
+
+	tables := []string{"users", "profiles", "educations", "certificates", "projects", "experiences", "achievements"}
```
Comment (OmkeshJosh) [inline internal/repository/profile_repository.go]: move to const file

File: internal/cron-job/app.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+)
+
+// InitCronJob initializes Cron Job for Backup purpose
+func InitCronJob(svc service.Service) {
+	zap.S().Info("Cron Job Initiated!")
+	c := cron.New()
+
+	// Add a cron job to run every midnight => "0 0 0 * * *"
+	// c.AddFunc("*/1 * * * *", func() { svc.BackupAllProfiles() })  FOR EVERY MINUTE BACKUP
+	c.AddFunc("0 0 0 * * *", func() { svc.BackupAllProfiles() }) //FOR EVERY MIDNIGHT BACKUP
```
Comment (OmkeshJosh) [inline internal/cron-job/app.go]: keep separate func

File: internal/db/migrate.go [hunk 3]
```diff
@@ -107,8 +124,152 @@ func (migration Migration) MigrationsDown() {
...
+			os.Remove(downMigrationFilePath)
+		}
+	}()
+
+	err = createFile(upMigrationFilePath)
+	if err != nil {
+		return
+	}
+
+	fmt.Printf("created %s\n", upMigrationFilePath)
```
Comment (OmkeshJosh) [inline internal/db/migrate.go]: use logger

File: internal/db/migrate.go [hunk 4]
```diff
@@ -107,8 +124,152 @@ func (migration Migration) MigrationsDown() {
...
+
+	err = createFile(downMigrationFilePath)
+	if err != nil {
+		return
+	}
+
+	fmt.Printf("created %s\n", downMigrationFilePath)
```
Comment (OmkeshJosh) [inline internal/db/migrate.go]: same for this

File: internal/db/migrate.go [hunk 5]
```diff
@@ -107,8 +124,152 @@ func (migration Migration) MigrationsDown() {
...
+	zap.S().Infof("latest migration version from local file system: %d", localDIRFileVersions[0])
+	return localDIRFileVersions[0], nil
+}
+
+func getMigrationVersionsFromDir(dir string) ([]int, error) {
+	return []int{}, nil
+}
+
+// GoTo migrates to a specific version
+func (migration Migration) GoTo(version uint) (err error) {
```
Comment (OmkeshJosh) [inline internal/db/migrate.go]: use meaningful function name.

File: internal/repository/profile_repository.go (hunk 1 above)
Comment (Abhishek-Jain-1925) [reply internal/repository/profile_repository.go]: Done!

File: internal/cron-job/app.go (hunk 2 above)
Comment (Abhishek-Jain-1925) [reply internal/cron-job/app.go]: Done!

File: internal/db/migrate.go (hunk 4 above)
Comment (Abhishek-Jain-1925) [reply internal/db/migrate.go]: Done!

File: internal/db/migrate.go (hunk 5 above)
Comment (Abhishek-Jain-1925) [reply internal/db/migrate.go]: Done!

File: internal/db/migrate.go (hunk 3 above)
Comment (Abhishek-Jain-1925) [reply internal/db/migrate.go]: Done!

**Comment by OmkeshJosh** [comment]
LGTM

**Pull request merged**